### ISSUE TITLE
update ts-scripts to build multi-platform images

### ISSUE THREAD

**Issue by sotojn** [terascope/teraslice#3522]
There is an issue with creating and publishing docker images in CI that only produce `amd/64` images in docker hub.

We need to look into the `dockerBuild()` function in [packages/scripts/src/helpers/scripts:line391](https://github.com/terascope/teraslice/blob/9a44c8eae4d1d41ceb96aab80478c197b9e0d09f/packages/scripts/src/helpers/scripts.ts#L391) and change [line:406](https://github.com/terascope/teraslice/blob/9a44c8eae4d1d41ceb96aab80478c197b9e0d09f/packages/scripts/src/helpers/scripts.ts#L406C1-L410C2) to use `docker buildx build` in the args section. 

We can then either hard code `--platforms linux/amd64, linux/arm64, darwin/amd64` inside the arguments or make it an option that gets passed in.

**Comment by godber** [comment]
When making this change we need to consider all of the use cases this build script gets used, including internal use cases, and make sure to provide the appropriate flexibility to handle those scenarios.

**Comment by sotojn** [comment]
I have implemented a solution in a PR #3527 that implements multiplatform builds of teraslice images in the case of a `merge to master` or a `release`. The issue is that the arm image builds very slow which may be tolerabole because CI only runs buildx in the cases stated above. It seems to be taking over 40 minutes to build the `yarn && yarn build` layer. I had to put a `--network-timeout` flag in yarn to not get a network timeout error as shown here in CI.

https://github.com/terascope/teraslice/actions/runs/7576194489/job/20634457971#step:6:1871

Here where I got the yarn network timeout solution:
https://github.com/yarnpkg/yarn/issues/8242

We could stick with having to deal with long builds or there are a couple other solutions.
- **look at ARM options for Github Runners**
  - It seems its in a closed beta right now, but it could be a potential solution later by building on two different runners with their native platforms and then merging a manifest.
  - https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/
- **look at the cross compilation Docker build option**
  - Instead of only using emulation to build, we can introduce cross compilation inside the docker file itself
  - Basically emulation copies the whole system of another architecture in software, while cross-compilation uses native binaries to create new ones for a different architecture. Cross-compilation is mainly for compiling from what I've read (hence the name maybe). They can be used together, like using emulation for installing packages and cross-compilation for building code in a Dockerfile. Which results in faster build times hopefully.
  - https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

**Comment by sotojn** [comment]
After doing a bunch of research and testing it on my test repo I decided we should just use the [`docker/build-push-action`](https://github.com/docker/build-push-action).

I looked at two other options before landing here:

- [docker/github-builder](https://github.com/docker/github-builder) is Docker's reusable workflow but it doesn't support `provenance: false` or custom computed tags, which we need for the `nodev{version}` tag format.
- [docker/metadata-action `type=raw`](https://github.com/docker/metadata-action?tab=readme-ov-file#typeraw) can generate custom tags but it adds an extra layer of indirection for something we can handle in a single shell line.

## final proposal
We build `amd64` and `arm64` each built natively on their own runner and push to GHCR by content digest (no tag yet). A third job then combines both digests into a single multi-arch manifest under the real tag. No platform suffixes, no stripping, no emulation. I made a lazy github pipeline mermaid diagram of before and after.


**Before:**

```mermaid
flowchart LR
    A[build-release\nts-scripts publish docker\namd64 only]
```

**After:**

```mermaid
flowchart LR
    A[build-docker-amd64\nubuntu-latest] --> C[create-docker-manifest]
    B[build-docker-arm64\nubuntu-24.04-arm] --> C
    C --> D[append-docker-tags-to-release]
```
